s3-vfs/src/main/java/org/pentaho/s3common/S3CommonFileObject.java

```diff
@@ -224,10 +224,21 @@ public void doAttach() throws Exception {
           injectType( FileType.IMAGINARY );
           return;
         }
+
+        // One common case is "403 Access Denied", for bucket names which exist in the same region
+        // but are not of this account. This can also happen for normal files and is being handled
+        // similarly in handleAttachExceptionFallback.
+        // Any other errors should also bubble up.
+
+        // Make sure this gets printed for the user.
+        logger.error( "Could not get information on " + getQualifiedName(), e );
+        throw new FileSystemException( "vfs.provider/get-type.error", getQualifiedName(), e );
```

Comment:
Also, please check if we will introduce any security issue by disclosing the exact error code and `getQualifiedName()` which includes the bucket name and a file path segment, in the message.

https://github.com/pentaho/big-data-plugin/blob/70b29e5b05692629d650f70822a8064939a28bc7/s3-vfs/src/main/java/org/pentaho/s3common/S3CommonFileObject.java#L418-L424

Reply:
AFAICT, the arguments to `FileSystemException` following the error code are not used to build the exception message. The exception message is directly the error code.

These additional arguments provide context for the exception for any callers that recognize the error code and know which arguments are passed as context. Again, don't think any callers are using these.

Despite this, I belieive that Spoon does typically allow the user to see the stack trace of exceptions (in some "Detailed error information..." option in the error UI).

Also, as pointed out, this 403 error is being handled for normal files in the same way in `handleAttachExceptionFallback`.

Also, not saying this justifies it, but I believe there are many cases out there of Apache VFS exceptions which include the path of a file in the contextual arguments.

Reply:
I had misread Apache VFS code regarding FileSystemException  and how these are formatted.
- [https://github.com/apache/commons-vfs/blob/4fbaade088e67d2f77215cec55631d171884de0[…]/src/main/java/org/apache/commons/vfs2/FileSystemException.java](https://github.com/apache/commons-vfs/blob/4fbaade088e67d2f77215cec55631d171884de0f/commons-vfs2/src/main/java/org/apache/commons/vfs2/FileSystemException.java#L198)
- [https://github.com/apache/commons-vfs/blob/4fbaade088e67d2f77215cec55631d171884de0[…]s-vfs2/src/main/java/org/apache/commons/vfs2/util/Messages.java](https://github.com/apache/commons-vfs/blob/4fbaade088e67d2f77215cec55631d171884de0f/commons-vfs2/src/main/java/org/apache/commons/vfs2/util/Messages.java#L30)
- [https://github.com/apache/commons-vfs/blob/4fbaade088e67d2f77215cec55631d171884de0[…]src/main/resources/org/apache/commons/vfs2/Resources.properties](https://github.com/apache/commons-vfs/blob/4fbaade088e67d2f77215cec55631d171884de0f/commons-vfs2/src/main/resources/org/apache/commons/vfs2/Resources.properties#L19)

I had assumed there was no actual construction of messages from the arguments from the first pass at reading the code…
This also tells me we shouldn’t be throwing these exceptions with codes which don’t already exist…